yt_dlp/plugins.py

```diff
@@ -0,0 +1,152 @@
+import importlib
+import sys
+import traceback
+from contextlib import suppress
+from importlib.abc import MetaPathFinder, Loader
+from importlib.machinery import ModuleSpec
+from importlib.util import module_from_spec, find_spec
+from inspect import getmembers, isclass
+from itertools import accumulate
+from pathlib import Path
+from pkgutil import iter_modules as pkgutil_iter_modules
+from shutil import copytree
+from zipfile import ZipFile
+from zipimport import zipimporter
```

Comment:
This made the code so much harder for me to read. It is better to import the modules `import importlib` and call the functions as `importlib.util.find_spec`, especially since u r declaring other similar names in the code.